src/shared/services/schema-validation.service.ts

```diff
@@ -21,40 +21,71 @@
 */
 
 import { Injectable } from '@angular/core';
+import * as Ajv from 'ajv';
 
 @Injectable()
 export class SchemaValidationService {
 
-  /**
-   * Validate string value against schema
-   *
-   * @throws {Error} if value can not be validated
-   *
-   * Checks: pattern
-   *
-   */
-  validateStringValue(value: string, schema: Object) {
-    let pattern = schema['pattern'];
-    if (pattern && !value.match(pattern)) {
-      throw Error(`Does not match with pattern: ${pattern}`);
-    }
+  private ajv = new Ajv();
+  // https://gist.github.com/dperini/729294
+  private reWebUrl = new RegExp(
```

Comment:
It covers the most cases based on a blog that I found...everything is a big word :p